src/index.ts

```diff
@@ -104,7 +112,10 @@ export async function prepare(
 
     const replaceInFileConfig = replacement as ReplaceInFileConfig;
 
-    replaceInFileConfig.to = template(replacement.to)({ ...context });
+    replaceInFileConfig.to =
+      typeof replacement.to === "function"
```

Comment:
This will never be a function as there is no parsing of the string to determine if it is a function. How do we get around calling `eval` here? https://github.com/semantic-release/semantic-release/blob/master/lib/plugins/utils.js#L54

Reply:
In the `release.config.js` we are passing in a function instead of a string like this:

```
module.exports = {
  branches: [{ name: 'main' }, { name: 'beta', prerelease: true }],
  plugins: [
    '@semantic-release/commit-analyzer',
    '@semantic-release/release-notes-generator',
    [
      '@google/semantic-release-replace-plugin',
      {
        replacements: [
          {
            files: ['app.config.js'],
            from: "version: '.*'", // eslint-disable-line
            to: "version: '${nextRelease.version}'", // eslint-disable-line
          },
          {
            files: ['app.config.js'],
            from: "buildNumber: '.*'", // eslint-disable-line
            to: "buildNumber: '${nextRelease.version}'", // eslint-disable-line
          },
          {
            files: ['app.config.js'],
            from: `versionCode: [^,]*`, // eslint-disable-line
            to: (match) => `versionCode: ${parseInt(match.split(':')[1].trim()) + 1}`, // eslint-disable-line
          },
        ],
      },
    ],
    '@semantic-release/changelog',
    '@semantic-release/npm',
    '@semantic-release/github',
    [
      '@semantic-release/git',
      {
        message: 'chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}',
        assets: ['CHANGELOG.md', 'package.json', 'package-lock.json', 'app.config.js'],
      },
    ],
  ],
};
```

Reply:
We have my fork installed in our package currently and it is correctly using the function syntax to update the `versionCode`.

Reply:
Can you add a note then specifying that `to` can only be a function if the js version of the config is used. I forgot about that. https://semantic-release.gitbook.io/semantic-release/usage/configuration#configuration-file